pkg/timelock/timelock.go

```diff
@@ -132,63 +132,81 @@ func (tw *Worker) Listen(ctx context.Context) error {
 		return err
 	}
 
+	tw.startLog()
+
+	// Handle OS signals to properly stop/kill the process.
 	stopCh := make(chan string)
-	logCh := make(chan types.Log)
-	sigCh := make(chan os.Signal, 1)
-	signal.Notify(sigCh, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM)
+	go handleOSSignal(stopCh)
 
+	// Update timelock-worker default scheduler delay.
 	tw.updateSchedulerDelay(time.Duration(tw.pollPeriod) * time.Second)
+
+	// Run the scheduler to add/del operations in a thread-safe way.
```

Comment:
the run scheduler doesn't have any way of being stopped by the Listen call.  it should either be cancellable by the passed context or listen to the created stop channel.  Right now it will leak